### PR TITLE
Provide README for AWSNativeSDK Libraries. Add instructions about how to add new service targets

### PR DESCRIPTION
**Pull request by onecent1101**
Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: package-system/AWSNativeSDK/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
++#### ElasticSearch ####
++ly_declare_aws_library(
++    NAME
++        ElasticSearch
++    LIB_FILE
++        aws-cpp-sdk-es
++)
+```
+It is for consumer to pull aws-cpp-sdk-es shared and static libraries as build and runtime dependencies by using `3rdParty::AWSNativeSDK::ElasticSearch` target.
```
Comment (allisaurus) [inline package-system/AWSNativeSDK/README.md:55]: nit: "It is up to the consumer to..." sounds more natural.

**Review by allisaurus — APPROVED** [review]

File: package-system/AWSNativeSDK/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Build New AWS Service Target in AWSNativeSDK
+First check package directories at [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) on GitHub. 
+The package name is the suffix of the directory name for the service.
+
+```
+aws-sdk-cpp\aws-cpp-sdk-<packageName>   # Repo directory name and packageName
+aws-sdk-cpp\aws-cpp-sdk-s3              # Example: Package name is s3
+```
+
+Note: If your target service is not available in current AWSNativeSDK version (but it is listed in aws-sdk-cpp repo),
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:10]: Nit: When we mention AWSNativeSDK thats really an internal term, consider linking to what this is or add a clarifying statement.

File: package-system/AWSNativeSDK/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Build New AWS Service Target in AWSNativeSDK
+First check package directories at [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) on GitHub.
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:2]: Maybe, "Find and identify the service package required from the package directory at ..."?

File: package-system/AWSNativeSDK/README.md (hunk 2 above)
Comment (lmbr-pip) [reply package-system/AWSNativeSDK/README.md:10]: Maybe also "If the AWS service packge is not currently used in AWSNativeSDK, then..."

File: package-system/AWSNativeSDK/README.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+Note: If your target service is not available in current AWSNativeSDK version (but it is listed in aws-sdk-cpp repo),
+then please upgrade AWSNativeSDK version first.
+
+## Example: Build ElasticSearch (ES)
+
+### 1. Check [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) for aws-cpp-sdk-es
+
+### 2. Add ElasticSearch as part of AWSNativeSDK build target
+Modify AWSNativeSDK build script by adding ElasticSearch as a build target, use windows build script as a reference:
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:18]: Nit: "Modify the AWSNativeSDK build script by adding ElasticSearch as a build target. The examples below use the windows build script as a reference:"

File: package-system/AWSNativeSDK/README.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            -DTARGET_ARCH=WINDOWS ^
+            -DCMAKE_CXX_STANDARD=17 ^
+            -DCPP_STANDARD=17 ^
+-           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
++           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;es;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
+            -DENABLE_TESTING=OFF ^
+            -DENABLE_RTTI=ON ^
+            -DCUSTOM_MEMORY_MANAGEMENT=ON ^
+```
+aws-cpp-sdk-es shared and static libraries will be generated after AWSNativeSDK build.
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:34]: "after AWSNativeSDK is built"?

File: package-system/AWSNativeSDK/README.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
++    NAME
++        ElasticSearch
++    LIB_FILE
++        aws-cpp-sdk-es
++)
+```
+It is for consumer to pull aws-cpp-sdk-es shared and static libraries as build and runtime dependencies by using `3rdParty::AWSNativeSDK::ElasticSearch` target.
+
+### 4. Update AWSNativeSDK with correct version and revision
+Modify build config and list file to bump AWSNativeSDK with correct version and revision value, use windows build as a reference:
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:58]: Not sure if are covering the difference between providing a new version or providing a new revision well. Check that we have enough guidance about when we set the version vs when to set the revision

**Review by lmbr-pip — COMMENTED** [review]
LG. Thanks for doing this. 

Can we change PR title so it states something like "Provide README for AWS C++ SDK Libraries. Add instructions about how to add new targets". Clear PR titltes make it easy for folks to find / understand what this PR is for without having to open it. Will help future maintainers.

File: package-system/AWSNativeSDK/README.md (hunk 1 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md]: done

File: package-system/AWSNativeSDK/README.md (hunk 2 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md]: done

File: package-system/AWSNativeSDK/README.md (hunk 4 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md]: done

File: package-system/AWSNativeSDK/README.md (hunk 5 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md]: done

File: package-system/AWSNativeSDK/README.md (hunk 6 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md]: This format is not for AWSNativeSDK package only, user should have enough guidance by onboarding this 3p system.

File: package-system/AWSNativeSDK/README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
+# Build New AWS Service Target in AWSNativeSDK
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:1]: I would add a line about the intent of the README.md, it jumps straight into adding a new AWS Service Target.

# Overview
This package defines and builds the [AWS C++ SDK]() libarires for O3DE. It builds all core libraries and an opinionated subset of AWS client libraries. To add a new AWS Service client (or target) please use the instructions below.

File: package-system/AWSNativeSDK/README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Find and identify the service package required from the package directory at [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) on GitHub.
+The package name is the suffix of the directory name for the service.
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/README.md:3]: nit: Add a note that several AWS services vend multiple client libraries, use the AWS C++ instructions to identify the client target(s) for the AWS Service.

File: package-system/AWSNativeSDK/README.md (hunk 7 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md:1]: will do

File: package-system/AWSNativeSDK/README.md (hunk 8 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md:3]: Can you clarify on this "use the AWS C++ instructions to identify the client target(s) for the AWS Service"?
Agree that some target may require multiple ones, but even with this note I don't know what is the instruction here?

**Review by junbo75 — APPROVED** [review]

File: package-system/AWSNativeSDK/README.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+```diff
+--- a/package-system/AWSNativeSDK/build_AWSNativeSDK_windows.cmd
++++ b/package-system/AWSNativeSDK/build_AWSNativeSDK_windows.cmd
+@@ -58,7 +58,7 @@ call cmake -S %SRC_PATH% -B %BLD_PATH%\%BUILD_TYPE%_%LIB_TYPE% ^
+            -DTARGET_ARCH=WINDOWS ^
+            -DCMAKE_CXX_STANDARD=17 ^
+            -DCPP_STANDARD=17 ^
+-           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
++           -DBUILD_ONLY="access-management;cognito-identity;cognito-idp;core;devicefarm;dynamodb;es;gamelift;identity-management;kinesis;lambda;mobileanalytics;queues;s3;sns;sqs;sts;transfer" ^
+            -DENABLE_TESTING=OFF ^
```
[the remark this answers is not in the document]
Comment (onecent1101) [reply package-system/AWSNativeSDK/README.md:29]: At top, I have provided example for what value user should look for in aws cpp sdk package list

File: package-system/AWSNativeSDK/README.md (hunk 8 above)
Comment (lmbr-pip) [reply package-system/AWSNativeSDK/README.md:3]: Maybe: "Find and identify the service package(s) required from the package directory at [aws/aws-sdk-cpp](https://github.com/aws/aws-sdk-cpp) on GitHub. Refer to the [AWS SDK for C++ Developer Guide](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/welcome.html) for further help in selecting the packages by service." ?

**Pull request merged**